ivy/stateful/module.py

```diff
@@ -1141,48 +1153,468 @@ def build_mode(self):
         return self._build_mode
 
     @property
-    def built(self):
+    def built_(self):
         return self._built
 
-    # Methods #
-    # ------- #
+    # Module Converters #
+    # ------------------#
+    
+    @staticmethod
+    def static_to_haiku_module(ivy_module):
+        class MyHaikuModel(hk.Module):
+            def __init__(self):
+                super(MyHaikuModel, self).__init__()
+
+            def __call__(self, *args, **kwargs):
+                ivy_module.v = ivy_module.v.map(
+                    lambda x, kc: hk.get_parameter(
+                        name=kc,
+                        shape=x.shape,
+                        dtype=x.dtype,
+                        init=lambda shape, dtype: ivy.to_native(ivy_module.v[kc])
+                    )
+                )
+                a, kw = ivy.args_to_native(*args, **kwargs)
+                ret = ivy_module._forward(*a, **kw)
+                if isinstance(ret, tuple):
+                    return ivy.args_to_native(*ret)
+                return ivy.to_native(ret)
 
-    def to_torch_module(self):
+        return MyHaikuModel
+
+    def to_haiku_module(self):
+        """
+        Converts an ivy Module instance to a Haiku Module instance.
+        Parameters
+        ----------
+        ivy_module
+            The ivy module instance to convert
+        Returns
+        -------
+        ret
+            The new trainable hk.Module instance.
+        """
+
+        return ivy.Module.static_to_haiku_module(self)
+
+    def to_keras_module(self):
+        """
+        Converts an ivy Module instance to a Keras Module instance.
+        Parameters
+        ----------
+        self
+            The ivy module instance to convert
+        Returns
+        -------
+        ret
+            The new trainable tf.keras.Module instance.
         """
-        Convert a trainable ivy.Module instance to an instance of a trainable torch
-        module.
 
+        return MyTFModule(self)
+
+    def to_torch_module(self):
+        """
+        Converts an ivy Module instance to a Torch Module instance.
         Parameters
         ----------
         self
-            trainable ivy.Module instance
+            The ivy module instance to convert
+        Returns
+        -------
+        ret
+            The new trainable torch.nn.Module instance.
+        """
+
+        return MyTorchModule(self)
+
+    @staticmethod
+    def from_haiku_module(
+            native_module,
+            args=None,
+            kwargs=None,
+            device=None,
+            devices=None,
+    ):
+        """
+        Converts a Haiku module instance to an Ivy module instance.
```

Comment:
Because of the fact that both a native class as well as an instance of a native class can be passed to the `from_<backend>_module()` methods, I had to follow the convention that constructor parameters need to be keyword only whereas forward pass parameters need to be position only (when passing a class rather than an instance). This is necessary in order to avoid conflicts caused when propagating the `*args` and `**kwargs` through different methods of the Module class internally.

Reply:
because both the class and the class instance can receive arbitrary `args` and `kwargs`, a better approach would be to have something like `constructor_args`, `constructor_kwargs`, `instance_args`, `instance_kwargs` all as keyword arguements, without using `*` in the arguments to the outermost converter function. I haven't dived deeper and the argument names could be different, but in general if there are multiple targets for different `args` and `kwargs` then it's better to avoid `*` when catching them in the outer function arguments, if that makes sense?